ksqldb-execution/src/main/java/io/confluent/ksql/execution/interpreter/TermEvaluationContext.java

```diff
@@ -31,11 +31,11 @@
   private final Deque<Map<String, Object>> variableMappingsStack = new LinkedList<>();
 
   public TermEvaluationContext(final GenericRow row) {
-    this.row = row;
+    this.row = GenericRow.genericRow(row.values().toArray());
```

Comment:
Make a deep-copy. Similar, below.

Reply:
this might have real performance implications for making a deep copy per-row on each context evaluated

\cc @AlanConfluent to confirm

Reply:
If there are concerns, we can also suppress here (and maybe do a follow up refactoring is suitable?).

Reply:
👍